mmv1/third_party/terraform/resources/resource_bigtable_gc_policy.go

```diff
@@ -324,6 +351,55 @@ func generateBigtableGCPolicy(d *schema.ResourceData) (bigtable.GCPolicy, error)
 	return policies[0], nil
 }
 
+func getGCPolicyFromJSON(p map[string]interface{}) (bigtable.GCPolicy, error) {
```

Comment:
Question: Do we want to check if only allowed field is set in `gc_rule`?

I tested it locally with `gc_rules = "{\"mode\":\"union\",\"rules\":[{\"invalid_input\":10},{\"mode\":\"intersection\",\"rules\":[{\"max_age\":\"2h\"},{\"max_version\":3}]}]}"` and looks like the policy is created successfully without the invalid field and corresponding `mode` and it did not throw any errors. 
The policy is created as 
```
"gcRule": {
  "intersection": {
    "rules": [
      {
        "maxAge": "7200s"
      },
      {
        "maxNumVersions": 3
      }
    ]
  }
}
```
 Is it to be handled this way intentionlly? My concern is that users may accidentally include some typos (such as use `max_versions` instead of `max_version`) and the resource will be created successfully without users being aware of that it was actually not in the way they configured.

Reply:
I agree. It should be:

parsed_rules = ...
verify number of rules against mode.

sanity check:
- number of elements in rules ==  parsed_rules.
- also verify only have "mode" and "rules".

Reply:
A worst case scenario can be like this:

mode = "intersection"
rules = [
  max_ag = .   // the user made a typo here.
  max_version = .
  mode = .
]
 
Now, the user may think that we have a max_ag[e] & max_version & mode, but we silently created max_version & mode.

Reply:
Will add the check today, thanks for checking!

Reply:
Added validation code and more test.